src/App.tsx

```diff
@@ -16,80 +17,139 @@ export const goodsFromServer = [
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((prevGood, nextGood) => {
+    switch (sortType) {
+      case 'none':
+        return 0;
+
+      case 'alphabet':
+        return prevGood.localeCompare(nextGood);
+
+      case 'length':
+        return prevGood.length - nextGood.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  resetGoodsList = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  reverseGoodsList = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const {
+      sortByAlphabet,
+      sortByLength,
+      resetGoodsList,
+      reverseGoodsList,
+    } = this;
+    const isReseted = sortType === 'none' && !isReversed;
+    const listOfGoods = getReorderedGoods(
+      goodsFromServer,
+      { isReversed, sortType },
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info is-light', {
+              'is-light': sortType !== 'alphabet',
+            })}
+            onClick={sortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success', {
+              'is-light': sortType !== 'length',
```

Comment:
You have always the same classes
![image](https://user-images.githubusercontent.com/91411692/233417369-c48e5113-0495-4384-b7b2-7241e67bc927.png)
![image](https://user-images.githubusercontent.com/91411692/233417426-d81df202-0beb-4e43-ba7e-dbb3bed135a2.png)